pomegranate/NaiveBayes.pyx

```diff
@@ -320,3 +321,21 @@ cdef class NaiveBayes( object ):
                     y_ptr[i] = j
 
         return y
+
+    def to_json( self, separators=(',', ' : '), indent=4 ):
+        nb = {
+            'class' : 'NaiveBayes',
+            'models' : [ json.loads( model.to_json() ) for model in self.models ],
+            'weights' : self.weights.tolist()
+        }
+
+        return json.dumps( nb, separators=separators, indent=indent )
+
+    def from_json( cls, s ):
+        d = json.loads( s )
```

Comment:
Can you add in the ability to read from a file if a filename is given as well? Otherwise looks good to go